File: Java/src/com/dbc/algorithm/Sort/ArrayPartitionI.java
package com.dbc.algorithm.Sort;

import java.util.Arrays;

public class ArrayPartitionI {
    public static void quickSort(int[] nums, int left, int right) {
        if (left >= right) {
            return;
        }
        int l = left, r = right, mid = nums[left];
        while (l < r) {
            while (l < r && nums[r] >= mid) {
                r--;
            }
            if (l < r) {
                nums[l] = nums[r];
                l++;
            }
            while (l < r && nums[l] <= mid) {
                l++;
            }
            if (l < r) {
                nums[r] = nums[l];
                r--;
            }
        }
        nums[l] = mid;
        quickSort(nums, left, l - 1);
        quickSort(nums, l + 1, right);
    }

    public static int arrayPairSum(int[] nums) {
        Arrays.sort(nums);
        int result = 0;
        for (int i = 0; i < nums.length; i += 2) {
            result += nums[i];
        }
        return result;
    }
}

File: Python/sort/array_partition_i.py
from typing import List

def quick_sort(nums: List[int], left: int, right: int):
    if left >= right:
        return

    l, r = left, right
    mid = nums[left]
    while l < r:
        while l < r and nums[r] >= mid:
            r -= 1
        if l < r:
            nums[l] = nums[r]
            l += 1
        while l < r and nums[l] <= mid:
            l += 1
        if l < r:
            nums[r] = nums[l]
            r -= 1
    nums[l] = mid
    quick_sort(nums, left, l - 1)
    quick_sort(nums, l + 1, right)

def arrayPairSum(nums: List[int]) -> int:
    quick_sort(nums, 0, len(nums) - 1)
    result = 0
    for i in range(0, len(nums), 2):
        result += nums[i]
    
    return result

if __name__ == "__main__":
    print(arrayPairSum([6,2,6,5,1,2]))
        